### PR TITLE
create pylint action

### PR DESCRIPTION
**Pull request by soutobias**
I have established a 'pylint_checking.yml' configuration file to automate pylint checks on the GitHub actions.

After running pylint, the resulting code quality score is compared to the value stored in the '.pylint-score' file. If the newly calculated score exceeds the value stored in the file, the test will pass.

Subsequently, the value in the '.pylint-score' file is updated. This ensures that any new code additions maintain or improve upon the existing code quality score.

The current threshold was 5.0, but it was updated during this pull request to 5.58.

### CODE REVIEW

**Comment by soutobias** [comment]
https://github.com/British-Oceanographic-Data-Centre/COAsT/issues/550

File: .github/workflows/pylint_checking.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+name: Pylint
+on: pull_request
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9]
+    steps:
+      - uses: actions/checkout@v1
```
Comment (thogar-computer) [inline .github/workflows/pylint_checking.yml:11]: is there a reason for using v1 of the both the checkout and python actions
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/pylint_checking.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Set up Python ${{ matrix.python-version }} for Pylint
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Pylint
+        run: pip install pylint
+      - name: Check Score
+        run: |
+          echo "Start checking scores"
+          pylint `find . -name "*.py"` | grep 'Your code has been rated' | awk '{print $10}' | awk -F/ '{print $1}'
```
Comment (thogar-computer) [inline .github/workflows/pylint_checking.yml:21]: does this pylint all py files in the repo or just the one within the coast directory?

File: .github/workflows/pylint_checking.yml (hunk 1 above)
Comment (soutobias) [reply .github/workflows/pylint_checking.yml:11]: I try to use the same as the "formating.yml". However, I think we can change to v4

File: .github/workflows/pylint_checking.yml (hunk 2 above)
Comment (soutobias) [reply .github/workflows/pylint_checking.yml:21]: The pylint will run in all python files, including _init.py, config.py, tests, setup.py. It is normally the way I apply Pylint on my codes. However, I think we can change it to just the coast directory.

Comment (soutobias) [reply .github/workflows/pylint_checking.yml:21]: @thogar-computer , if you prefer, I can update the pull request with the pylint only for the coast directory

Comment (thogar-computer) [reply .github/workflows/pylint_checking.yml:21]: @soutobias thats ok. this can be closed

**Pull request merged**